torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -857,7 +864,12 @@ const char* WARN_RESIZE =
     " can't be represented in the JIT at the moment, so we won't connect any uses of "
     "this value with its current trace. If you happen to use it again, it will show "
     "up as a constant in the graph.";
-
+const char* STRICT_TRACER_MSG =
+    " might cause the trace to be incorrect, this is only valid if the container "
+    "structure (i.e. dict keys) does not change based on the module's inputs, consider "
+    "using constant container structure instead(i.e. Tuple, NamedTuple) to avoid this "
+    "warning/error, If you absolutely need this and know the side effects, pass "
+    "strict=False to trace() to allow this behavior.";
```

Comment:
This message talks exclusively about `dict`, but it is used when tracing a `list` output. Can we generalize this? Maybe something like:

```
" might cause the trace to be incorrect. This is only valid of the container structure does not change based on the module's inputs. Consider using a constant container instead (e.g. for `list`, use a `tuple` instead. for `dict`, use a `NamedTuple` instead). <...>
```